onnxruntime/core/providers/webgpu/math/gemm.cc

```diff
@@ -216,7 +217,8 @@ Status Gemm::ComputeInternal(ComputeContext& context) const {
   const uint32_t num_tile_n = (N + TILE_SIZE - 1) / TILE_SIZE;
   const uint32_t num_tile_m = (M + TILE_SIZE - 1) / TILE_SIZE;
 
-  program.AddOutputs({{Y, ProgramTensorMetadataDependency::Type}})
+  program.CacheHint(absl::StrJoin(std::make_tuple(alpha_, transA_, transB_), ","))
```

Comment:
You can directly write it as `program.CacheHint(alpha_, transA_, transB_)`. I think `CacheHint` underlying has used `absl::StrJoin`.